proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,236 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existentials-with-fixed-associated-types.md)
+* Authors: [Filip Sakel](https://github.com/filip-sakel), [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Today, Swift protocols are divided into two categories: those that _can_ and those that _can't_ be used as Types. The latter can only be used as generic constraints and _not_ as Types - because they have 'Self' or associated type requirements. However, in some cases where a protocol inherits and fixes the associated types of another protocol - that doesn't have `Self` requirements, this constraint seems rather unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.
```

Comment:
Sure! It's also more succinct which is important for the introduction. Moreover, I like the inclusion of the infamous error message. I'm just not sure about "[protocols that] _can_ be used to represent the type of an existential value". Your proposed text is more descriptive, but it also feels less comprehensible. I chose "_can_ be used as Types" so as to match the [Language Guide](https://docs.swift.org/swift-book/LanguageGuide/Protocols.html#ID275). I recognise that my version is not nearly as accurate as yours; I'm worried, though - that it might be confusing. Of course, it could be argued that since this is an evolution proposal discussing the future of Swift, then maybe refined terminology should be used. What do you think?

Reply:
The language guide is a good point. How about we say "those that can be used as fully fledged types"?